src/MonoTorrent/MonoTorrent.Client/Modes/InitialSeedingMode.cs

```diff
@@ -92,7 +92,7 @@ public override void Tick (int counter)
             if (unchoker.Complete) {
                 PeerMessage bitfieldMessage = new BitfieldMessage (Manager.Bitfield);
                 PeerMessage haveAllMessage = new HaveAllMessage ();
-                foreach (PeerId peer in Manager.Peers.ConnectedPeers) {
+                foreach (PeerId peer in Manager.Peers.ConnectedPeers.ToArray()) {
```

Comment:
Same here - ConnectedPeers cannot be mutated while this code is running (if it were possible then it could also mutate while `ToArray` is being run, which could result in similar issues).